packages/eslint-plugin-nx/src/rules/enforce-module-boundaries.ts

```diff
@@ -232,7 +233,19 @@ export default createESLintRule<Options, MessageIds>({
       // same project => allow
       if (sourceProject === targetProject) {
         // we only allow relative paths within the same project
-        if (!allowCircularSelfDependency && !isRelativePath(imp)) {
+        // and if it's not a secondary entrypoint in an angular lib
+        const ngPackageExistsForSecondaryEntry = sourceFilePath.endsWith(
```

Comment:
@meeroslav @leosvelperez This is a very specific fix for Angular. I'm not sure if the same problem exists with standard libraries as their secondary entry points are likely defined in the package.json. It might still exist.

There also may be a better way to solve this. Using a glob would be safer. I'm open to suggestions!

Reply:
I don't think check for src/index.ts is necessary. Project config might override this so it wouldn't work in every scenario anyway.

Reply:
Part of the problem is that project config knows nothing about the secondary entrypoint, so some assumptions have to be made. I'm making the assumption that the entry file is index.ts because that is what is generated by the secondary endpoint generator.

I'll move the logic into a util! :)

Reply:
I also cannot move it into util in `linter` package as it creates a circular dependency. 
I'll place it in `eslint-plugin-nx`